BladeportBinaryTreeManager.Web/wwwroot/app/src/app/user/user.component.ts

```diff
@@ -0,0 +1,87 @@
+import { Component, OnInit } from '@angular/core';
+import { UserService } from '../user/user.service';
+import { User } from '../user/user.model';
+
+@Component({
+  selector: 'app-users',
+  templateUrl: './user.component.html',
+  styleUrls: ['./user.component.css']
+})
+
+export class UserComponent implements OnInit {
+
+  userList: User[];  
+  input;
+  todaysDate: Date;
+  enableEdit = false;
+  isEditing = false;
+  enableEditIndex = null;
+
+  constructor(private _userService: UserService) { 
+    this.userList = [];
+
+    this.input = {
+      'userName':'',
+      'firstName': '',
+      'lastName': ''
+    };
+
+    this.todaysDate = new Date();
+  }
+
+  ngOnInit() {
+    this._userService.getUsers().subscribe((userList: User[]) => {
+      this.userList = userList;
+    });    
+  }
+
+  adduser() {    
+    this._userService.addUser(this.input)
+      .subscribe(() => {
+        this.ngOnInit(); 
```

Comment:
In practice, do not refer to the ngOnInit() event directly. When the application grows, and the OnInit needs to have additional procedures and/or variable assignment - it will not be useable to the calling method anymore. Create a new method that contains the ngOnInit() logic and call it from there.